contracts/strategies/BenqiStrategyForLP.sol

```diff
@@ -0,0 +1,204 @@
+        require(address(qiRewardToken) == swapPairToken.token1(), "Swap pair does not match QI reward");
+    }
+
+    function setAllowances() public override onlyOwner {
+        depositToken.approve(address(stakingContract), MAX_UINT);
+    }
+
+    function deposit(uint amount) external override {
+        _deposit(msg.sender, amount);
+    }
+
+    function depositWithPermit(uint amount, uint deadline, uint8 v, bytes32 r, bytes32 s) external override {
+        depositToken.permit(msg.sender, address(this), amount, deadline, v, r, s);
+        _deposit(msg.sender, amount);
+    }
+
+    function depositFor(address account, uint amount) external override {
+        _deposit(account, amount);
+    }
+
+    function _deposit(address account, uint amount) private onlyAllowedDeposits {
+        require(DEPOSITS_ENABLED == true, "BenqiStrategyForLP::_deposit");
+        if (MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST > 0) {
+            (uint avaxAmount, uint qiAmount, uint totalAvaxAmount) = _checkRewards();
+            if (totalAvaxAmount > MAX_TOKENS_TO_DEPOSIT_WITHOUT_REINVEST) {
+                _reinvest(avaxAmount, qiAmount);
+            }
+        }
+        require(depositToken.transferFrom(msg.sender, address(this), amount));
+        _stakeDepositTokens(amount);
+        _mint(account, getSharesForDepositTokens(amount));
+        totalDeposits = totalDeposits.add(amount);
+        emit Deposit(account, amount);
+    }
+
+    function withdraw(uint amount) external override {
+        uint depositTokenAmount = getDepositTokensForShares(amount);
+        if (depositTokenAmount > 0) {
+            _withdrawDepositTokens(depositTokenAmount);
+            _safeTransfer(address(depositToken), msg.sender, depositTokenAmount);
+            _burn(msg.sender, amount);
+            totalDeposits = totalDeposits.sub(depositTokenAmount);
+            emit Withdraw(msg.sender, depositTokenAmount);
+        }
+    }
+
+    function _withdrawDepositTokens(uint amount) private {
+        require(amount > 0, "BenqiStrategyForLP::_withdrawDepositTokens");
+        stakingContract.redeem(amount);
+    }
+
+    function reinvest() external override onlyEOA {
+        (uint avaxAmount, uint qiAmount, uint totalAvaxAmount) = _checkRewards();
+        require(totalAvaxAmount >= MIN_TOKENS_TO_REINVEST, "BenqiStrategyForLP::reinvest");
+        _reinvest(avaxAmount, qiAmount);
+    }
+
+    /**
+     * @notice Reinvest rewards from staking contract to deposit tokens
+     * @dev Reverts if the expected amount of tokens are not returned from `stakingContract`
+     * @param avaxAmount amount of WAVAX token to reinvest
+     * @param qiAmount amount of QI token to reinvest
+     */
+    function _reinvest(uint avaxAmount, uint qiAmount) private {
+        stakingContract.claimRewards();
+        uint amount = _reinvestToken(wavaxRewardToken, avaxAmount) + _reinvestToken(qiRewardToken, qiAmount);
```

Comment:
Nice solution, but this means that the reward distributed will be on both Qi and Wavax. One issue is how can we display this on the front-end, another issue is, how can the reinvest bots that exist out there have an interface to calculate the rewards. checkRewards on the other hand is returning the reward value as WAVAX. I don't have a strong though about this, I would prefer to keep it consistent, but there are other concerns that Yakman might want to address too

Reply:
Using WAVAX as a general rule in this scenario sounds good